radsat-sk/operation/subsystems/RBattery.c

```diff
@@ -0,0 +1,221 @@
+
+	i2c_received = batteryI2cTalk(batteryCurrentCommandBytes[2]);
+	dataStorage.outputCurrentBatteryBus = ADC_COUNT_TO_3V3_BUS_CURRENT_DRAW * i2c_received;
+
+
+	// Get ADC Temperatures
+	i2c_received = batteryI2cTalk(batteryTemperatureCommandBytes[0]);
+	dataStorage.motherboardTemp = (ADC_COUNT_TO_MOTHERBOARD_TEMP_SCALE * i2c_received)
+			- ADC_COUNT_TO_MOTHERBOARD_TEMP_SHIFT;
+
+	i2c_received = batteryI2cTalk(batteryTemperatureCommandBytes[1]);
+	dataStorage.daughterboardTemp1 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * i2c_received)
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	i2c_received = batteryI2cTalk(batteryTemperatureCommandBytes[2]);
+	dataStorage.daughterboardTemp2 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * i2c_received)
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	i2c_received = batteryI2cTalk(batteryTemperatureCommandBytes[3]);
+	dataStorage.daughterboardTemp3 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * i2c_received)
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	return dataStorage;
+}
+
+
+/**
+ * Send a telecommand via I2C to check on the current battery voltage, and if under 6.5V, raise the global safeFlag
+ * @return nothing.
+ */
+void checkSafeFlag(void){
+
+	// Create a temporary variable for receiving I2C data
+	uint16_t i2c_received;
+
+	// Get the battery output voltage
+	i2c_received = batteryI2cTalk(batteryVoltageCommandBytes[0]);
+	uint16_t converted_value = ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE * i2c_received;
+
+	// If the voltage is less than 6.5V then raise the safeFlag to send the cubeSat into safe mode.
+	// TODO: generate an error?
+	if(converted_value < 6.5){
+		safeFlag = 1;
+	}
+	else{
+		safeFlag = 0;
+	}
+}
+
+/***************************************************************************************************
+                                         PRIVATE FUNCTIONS
+***************************************************************************************************/
+
+static uint16_t batteryI2cTalk(uint32_t command) {
+
+	// Create a temporary variable for receiving I2C data
+	uint16_t i2c_data = 0;
+
+	// Create a temp variable for passing the proper command length
+	uint8_t comm_length;
+	uint8_t comm_delay;
+
+	// Variable selection for command length dependent on if it is a telemetry call
+	// Byte mask to look only at the command field to see if it is 0x10 (telem call)
+	if ((command && 0x0000FF0000) == 0x0000100000) {
+		comm_length = BATTERY_TELEM_COMMAND_LENGTH;
+		comm_delay = BATTERY_I2C_TELEM_DELAY;
+	}
+	else {
+		comm_length = BATTERY_COMMAND_LENGTH;
+		comm_delay = BATTERY_I2C_DELAY;
+	}
+
+	// tell Battery to give us data; and store into our internal buffer
+	int error = i2cTalk(BATTERY_I2C_SLAVE_ADDR, comm_length, BATTERY_RESPONSE_LENGTH,
+							command, i2c_data, comm_delay);
+
+	// return error? if an error occurs, else send the data back
+	// TODO: Make this return 0 on a failure? Printf the error instead? Get advice for this
```

Comment:
A similar comment was made in the eps PR. Review that PR first.

Wait until main cleanup is merged it will offer an example